colcon_cargo/task/cargo/test.py

```diff
@@ -31,29 +38,98 @@ async def test(self, *, additional_hooks=None):  # noqa: D102
         logger.info(
             "Testing Cargo package in '{args.path}'".format_map(locals()))
 
-        assert os.path.exists(args.build_base)
+        assert os.path.exists(args.build_base), \
+            'Has this package been built before?'
 
-        test_results_path = os.path.join(args.build_base, 'test_results')
-        os.makedirs(test_results_path, exist_ok=True)
+        test_results_path = os.path.join(args.build_base, 'cargo_test.xml')
 
         try:
             env = await get_command_environment(
                 'test', args.build_base, self.context.dependencies)
         except RuntimeError as e:
+            # TODO(luca) log this as error in the test result file
             logger.error(str(e))
             return 1
 
+        # Disable color to avoid escape sequences in test result file
+        env['NO_COLOR'] = '1'
```

Comment:
Does `--color never` in the cargo commands not work? Or is this colored output from somewhere else?

Reply:
From my experiments not really:

Stock invocation:

![image](https://github.com/user-attachments/assets/27e761fb-a1ce-4cdb-9d4a-3ce9ac910458)

Only `color=never`, note how it removes most of the colors:

![image](https://github.com/user-attachments/assets/900353a1-d146-41e3-a7d2-45c93cc4beb1)

Only `NO_COLOR=1`, removes the other set of colors:

![image](https://github.com/user-attachments/assets/910ccf9c-58a2-495a-b7b4-3540dcd20d8d)

Both flags, all colors are gone:

![image](https://github.com/user-attachments/assets/02945682-bece-4573-97e7-824f220852f0)

Reply:
`cargo test --color never -- --color never` seems to work 
![image](https://github.com/user-attachments/assets/49d82f67-1759-4a32-bcf0-ad3071dce126)